chargeOclock/Source/DataSources/ClientsDataSource.swift

```diff
@@ -11,13 +11,18 @@ import UIKit
 class ClientsDataSource: NSObject, UITableViewDataSource {
 
 	private(set) var clients: [Client]
+	var updateClient: ((Client) -> Void)?
 	let reuseIdentifier: String
 
 	init(with clients: [Client] = [], reuseIdentifier: String = "devClientCell") {
 		self.clients = clients
 		self.reuseIdentifier = reuseIdentifier
 	}
 
+	func reload(_ clients: [Client]) {
```

Comment:
why couldn't you just set the clients array?
and add a didSet handler if anything needs to be done if the array changes.